### PR TITLE
send notification on action failure

### PR DESCRIPTION
**Pull request by neha-Gupta1**
send a Slack notification if Github Actions fails

#### Does this PR need a docs update or release note?
- [ ] :no_entry: No

#### Type of change

- [ ] :robot: Supportability/Tests

#### Test Plan

- [ ] :muscle: Manual

### CODE REVIEW

**Comment by meain** [comment]
I've triggered a manual run on this branch at https://github.com/alcionai/corso/actions/runs/4729929001 just to see how this works.

File: .github/workflows/sanity-test.yaml [hunk 1]
```diff
@@ -364,4 +364,25 @@ jobs:
           path: src/testlog/*
           if-no-files-found: error
           retention-days: 14
-      
\ No newline at end of file
+      
+      - name: Send Github Action failure to Slack
```
Comment (meain) [inline .github/workflows/sanity-test.yaml:368]: Can we try a manual run for this to validate the integration?

Comment (neha-Gupta1) [reply .github/workflows/sanity-test.yaml:368]: We need to add slack-webhook in the secrets and then can test it. I will let you know once its added.

Comment (ryanfkeepers) [reply .github/workflows/sanity-test.yaml:368]: Could we add a similar action to the cleanup scripts?  Would be nice to know when those fail, too.  Thanks!

Comment (vkamra) [reply .github/workflows/sanity-test.yaml:368]: @neha-Gupta1 - created SLACK_WEBHOOK_URL secret

Comment (neha-Gupta1) [reply .github/workflows/sanity-test.yaml:378]: As Vaibhav mentioned the secret is added and I have done some testing too. Its ready for review now.

**Review by ryanfkeepers — APPROVED** [review]

**Pull request merged**